rfcs/013-introduce-headless-components-library.md

```diff
@@ -0,0 +1,50 @@
+# Introduce a headless (unstyled) components library
+
+## Affected projects
+
+- [Frontend](https://github.com/opencollective/opencollective-frontend)
+- [Frontend template](https://github.com/opencollective/opencollective-frontend-template) (since it depends on Frontend components)
+
+## Motivation
+
+- Reduce the load on the engineering team by re-using components already built with UX/accessibility best practices
+- Build new UI more easily, through a library of components that are already tested and documented
+- Prepare the ground for introducing a [CSS library](https://github.com/opencollective/opencollective/pull/6694)
+
+## Solution
+
+<!--
+Describe the proposed solutions with facts and arguments in favor of this change.
+If it has some drawbacks, include them as well.
+-->
+
+## Alternatives
+
+|                                   | Material Base                 | Headless UI | PrimeReact | Radix UI                         | ReaKit | Reach UI | React Aria |
+| --------------------------------- | ----------------------------- | ----------- | ---------- | -------------------------------- | ------ | -------- | ---------- |
+| Popularity (weekly npm downloads) | 3.3m                          | 1.1m        | 87k        | 1.9m (@radix-ui/react-primitive) | 165k   | n/a      | 163k       |
+| Maturity                          | In beta (introduced Sep 2022) |             |            | Mature (first release Dec 2020)  |        |          |            |
```

Comment:
About Material's maturity, their README indicates:

> NOTE: Material Components Web tends to release breaking changes on a monthly basis, but follows [semver](https://semver.org/) so you can control when you incorporate them. We typically follow a 2-week release schedule which includes one major release per month with breaking changes, and intermediate patch releases with bug fixes.

Reply:
Also, the React bindings are not mature at all; see https://github.com/material-components/material-components-web/blob/master/docs/framework-wrappers.md. Overall Material base doesn't seem production-ready.

Reply:
```suggestion
| Maturity                          | Recent release (introduced Sep 2022), frequent breaking changes, React bindings not production-ready |             |            | Mature (first release Dec 2020)  |        |          |            |
```